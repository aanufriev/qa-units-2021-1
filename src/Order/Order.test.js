jest.mock('../utils/getDate');

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react'
import Order from './Order';
import {fakeOrders} from '../data/fakeOrders';
import {getDate} from '../utils/getDate';

configure({adapter: new Adapter() });

describe('Order.js', () => {
  beforeEach(() => {
    getDate.mockReturnValue("1 января 2000");
  })

  afterEach(() => {
    jest.resetModules();
  })

  it('test ok', () => {
    const wrapper = shallow(<Order order={fakeOrders[0]}/>);

    expect(getDate).toHaveBeenCalledTimes(1);

    expect(wrapper).toMatchSnapshot();
  });

  it('test null order', () => {
    const wrapper = shallow(<Order order={null}/>);

    expect(getDate).toHaveBeenCalledTimes(1);

    expect(wrapper.getElement()).toBeNull();
  });

  it('test empty items', () => {
    let order = Object.assign({}, fakeOrders[0]);
    delete order.items;  
    const wrapper = shallow(<Order order={order}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('test empty shop', () => {
    let order = Object.assign({}, fakeOrders[0]);
    delete order.shop;

    const wrapper = shallow(<Order order={order}/>);

    expect(wrapper.getElement()).toBeNull();
  });

  it('test empty date', () => {
    let order = Object.assign({}, fakeOrders[0]);
    delete order.date;
    const wrapper = shallow(<Order order={order}/>);

    expect(wrapper.getElement()).toBeNull();
  });
});

