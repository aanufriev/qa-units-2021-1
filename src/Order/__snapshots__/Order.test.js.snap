// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js test empty items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      1 января 2000
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js test ok 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      1 января 2000
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;
